### PR TITLE
Missing attributes and improved summary for Cox models

### PR DESCRIPTION
**Pull request by kshedden**
Miscellaneous minor improvements to PHReg:

* attach `df_resid` and `df_model` attributes (otherwise can't be used with `predict_functional` and perhaps other places)

* better handling of variable names passed as auxiliary variables (e.g. `entry`)

* provide additional information about dropped strata and entry times in summary output

### CODE REVIEW

File: statsmodels/duration/hazard_regression.py [hunk 1]
```diff
@@ -324,12 +326,16 @@ def __init__(self, endog, exog, status=None, entry=None,
...
-                                    self.entry, self.offset)
+                                   self.exog, self.strata,
+                                   self.entry, self.offset)
 
         # TODO: not used?
         self.missing = missing
 
+        self.df_resid = (np.float(self.exog.shape[0] -
+                                  np_matrix_rank(self.exog)))
+        self.df_model = np.float(np_matrix_rank(self.exog) - 1)
```
Comment (josef-pkt) [inline statsmodels/duration/hazard_regression.py]: Is the minus 1 for the constant?
Is the constant implicit in the model?

**Comment by josef-pkt** [comment]
AFAIR, results.`t_test` requires df_resid even if it's not used.

I haven't removed that requirement from t_test, because I wasn't sure whether we want to make df_resid optional or keep it as a required attribute.

**Comment by josef-pkt** [comment]
TravisCI errors are unrelated to this PR.

File: statsmodels/duration/tests/test_phreg.py [hunk 2]
```diff
@@ -272,6 +272,20 @@ def test_summary(self):
         rslt = mod.fit()
         rslt.summary()
 
+        strata = np.kron(np.arange(50), np.ones(4))
+        mod = PHReg(time, exog, status, strata=strata)
+        rslt = mod.fit()
+        smry = rslt.summary()
+        msg = "3 strata dropped for having no events"
+        assert(msg in str(smry))
```
Comment (josef-pkt) [inline statsmodels/duration/tests/test_phreg.py]: numpy.testing `assert_`
plain assert are removed when byte compiling Python with -o or something like that

File: statsmodels/duration/hazard_regression.py (hunk 1 above)
Comment (kshedden) [inline statsmodels/duration/hazard_regression.py]: This was a mistake.  There's no intercept in a Cox model, so should not subtract 1.

**Comment by josef-pkt** [comment]
looks ready to merge to me

**Pull request merged**